2-WebApp-graph-user/2-5-HybridFlow/wwwroot/js/site.js

```diff
@@ -0,0 +1,24 @@
+﻿const msalInstance = new msal.PublicClientApplication({
+    auth: {
+        clientId: '[YOUR CLIENT ID]',
+        redirectUri: 'https://localhost:7089/',
+        authority: 'https://login.microsoftonline.com/[YOUR DOMAIN OR TENANT ID HERE]',
+        postLogoutRedirectUri: 'https://localhost:7089/',
+    },
+    cache: {
+        cacheLocation: 'sessionStorage',
+        storeAuthStateInCookie: false,
+    }
+});
+
+async function signOut() {
+    msalInstance.logoutRedirect({
+        onRedirectNavigate: _ => undefined
+    });
+
+    fetch('/logout', {
+        method: 'POST',
+        mode: 'same-origin',
+        credentials: 'same-origin',
+    });
+}
```

Comment:
Simple JavaScript to setup the MSAL client and sign out functionality to make it available to the application.

The msalInstance acts as a globally available object to the site here and is not too interesting.

The signOut method does two important things.

First, it triggers a client-side logout destroying all tokens in the client side cache.

Then, it sends a signout request to the server which empties the server cache and clears the session.

I'm unsure if there would be a case where multiple users would cache access tokens in the same browser profile but can look into this.